[TimothyLKM] ip

TimothyLKM opened the pull request.

Commit:
Change class name
Add intro and exit message

Commit:
Completed level 1- Echo

Commit:
Add a toDoList to track task
Add additional methods to print list

Commit:
Add new class named Task
Add methods to mark and unmark tasks

Commit:
Add new class named Task
Add methods to mark and unmark tasks

vimalapugazhan commented on src/main/java/TimL.java:
```diff
@@ -0,0 +1,68 @@
+import java.util.List;
+import java.util.Scanner;
+import java.lang.String;
+
```
Not sure if you need all these imports

vimalapugazhan commented on src/main/java/TimL.java:
```diff
@@ -0,0 +1,68 @@
+            addToList(text, list);
+        }
+    }
+
```
Could consider refactoring the triple nested if statements

vimalapugazhan commented on src/main/java/TimL.java:
```diff
@@ -0,0 +1,68 @@
+                "    added: " + task + " to the list\n" +
+                "    ____________________________________________________________\n");
+    }
+
```
Could consider putting the the lines and greetings as static variable instead of typing out every time

vimalapugazhan commented on src/main/java/TimL.java:
```diff
@@ -0,0 +1,68 @@
+                "     Bye. Hope to see you again soon!\n" +
+                "    ____________________________________________________________\n");
+    }
+}
```
The naming and indentations are good overall

vimalapugazhan reviewed (commented):
Code generally follows coding and naming standards and follows camel case. Other than your in else f=nesting the code looks good!

EugeneChanJiajun commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,29 @@
+    }
+    public void markAsComplete() {
+        this.status = true;
+        System.out.println("    ____________________________________________________________\n" +
```
Can try to separate some of the code as their own methods like the printing of separation lines

EugeneChanJiajun commented on src/main/java/TimL.java:
```diff
@@ -0,0 +1,68 @@
+    public static void printToDoList(List<Task> list){
+        System.out.println("    ____________________________________________________________");
+        System.out.println("     Here are the tasks in your list:");
+
```
Can try to close out the random blank lines to improve code readability

Commit:
Refactor if else loops into a switch
Add task manager, printer, text parser
Add Events, todo and deadline classes
Reorganise main

Commit:
Merge branch 'master' of https://github.com/TimothyLKM/ip

# Conflicts:
#	src/main/java/Task.java
#	src/main/java/TimL.java

Commit:
Remove unintended merges

Commit:
Implement new exceptions

sevenseasofbri commented on src/main/java/timl/Exeptions/EmptyException.java:
```diff
@@ -0,0 +1,4 @@
+package timl.Exeptions;
```
Names representing packages should be in all lower case.

https://se-education.org/guides/conventions/java/basic.html#naming

sevenseasofbri commented on src/main/java/timl/TimL.java:
```diff
@@ -0,0 +1,104 @@
+import java.lang.String;
+
+public class TimL {
+    public static void respondToCommand(String response, Task[] list){
```
this function is quite long (30+ LOC). this can affect redability. consider abstracting out parts of the code into relevant functions

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods

sevenseasofbri commented on src/main/java/timl/TimL.java:
```diff
@@ -0,0 +1,104 @@
+                try {
+                    taskIndex = Integer.parseInt(message) - 1;
+                    TaskManager.mark(taskIndex, list, emptyIndex);
+                }catch (NumberFormatException | TimException e){
```
ensure the spacing for the start of the catch block is there. the layout of a try catch block is as follows:

```java
try {
    statements;
} catch (Exception exception) {
    statements;
}
```
try maintaining this spacing

sevenseasofbri commented on src/main/java/timl/TimL.java:
```diff
@@ -0,0 +1,104 @@
+
+    public static void printList(Task[] list, int index){
+        Printer.printList();
+        int i = 1;
```
a more descriptive name would be helpful here 

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain

sevenseasofbri commented on src/main/java/timl/TimL.java:
```diff
@@ -0,0 +1,104 @@
+
+
+    public static void main(String[] args) {
+        Task[] todolist = new Task[100];
```
avoid having magic literals in the code. consider declaring a class-level constant instead

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers

sevenseasofbri commented on src/main/java/timl/utility/Printer.java:
```diff
@@ -0,0 +1,83 @@
+package timl.utility;
+
+public class Printer {
+    protected static final String GREETING = "    ____________________________________________________________\n" +
```
good use of constants

sevenseasofbri reviewed (commented):
good job overall👍🏽

Commit:
Change list to ArrayList
Add delete function to the task manager

Commit:
Add FileManager
Add FileParser

Commit:
Add method in TaskManager getTaskType

Commit:
Merge branch 'branch-Level-6'

# Conflicts:
#	src/main/java/timl/TimL.java
#	src/main/java/timl/task/TaskManager.java
#	src/main/java/timl/utility/Printer.java

Resolve merge conflicts

Commit:
Resolve conflicts
Fix file export issues

Commit:
Change file parser divider

Commit:
Branch-Level-9
Give users a way to find a task by searching for a keyword.

Commit:
A-JavaDoc
Add JavaDoc comments

TimothyLKM committed:
Merge pull request #1 from TimothyLKM/branch-Level-9

Branch-Level-9

Commit:
Resolve merge conflicts after merging master into branch

TimothyLKM committed:
Merge pull request #2 from TimothyLKM/branch-A-JavaDoc

merge Branch A-Java-Doc

Commit:
MoreOOP
make the code more OOP with addition of command handler

Commit:
A-UserGuide
